src/sentry/workflow_engine/handlers/detector/stateful.py

```diff
@@ -358,20 +377,22 @@ def evaluate(
         condition_evaluation, new_priority = self._evaluation_detector_conditions(value)
         state = self.state_manager.get_state_data([None])[None]
 
-        if state.status == new_priority or not condition_evaluation:
+        if (state.status == new_priority) or (  # The status hasn't changed
+            not condition_evaluation
+        ):  # If the condition is not met
```

Comment:
Nit: I think this would be easier to read like
```
if (
    state.status == new priority  # The status hasn't changed
    or not condition_evaluation # The condition is not met
)
```

Reply:
🤣 that's how i had it and our linter rewrote it to this 😭

Reply:
I had a feeling it was some kind of automation that did this!

Reply:
I would extract them to variables then

```py
if not status_changed or not evaluation_met:
```

Reply:
👍 - i had added these comments when i was deving and had 3 conditions as well, i ended up rewriting it a little bit, i think it reads a bit cleaner now (will also be refactoring this a little further after this PR is merged)